### PR TITLE
218 update donation step

### PR DESCRIPTION
**Pull request by L03TJ3**
# Description
Updating donation page with new design.

Todo:
- [x] - add estimation calculations for streaming
- [x] - mobile responsiveness

About # (link your issue here)
#218 

## Design
image

Screenshot from 2024-11-26 14-14-48

Screenshot from 2024-11-26 10-39-21

### CODE REVIEW

**Comment by sirpy** [comment]
@L03TJ3 please attached images of different states

**Review by sirpy — CHANGES_REQUESTED** [review]

File: packages/app/src/models/constants.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import GdContracts from '@gooddollar/goodprotocol/releases/deployment.json';
 import env from '../lib/env';
 
 // 5%
-export const acceptablePriceImpact = 5;
```
[the remark this answers is not in the document]
Comment (L03TJ3) [reply packages/app/src/models/constants.ts:6]: requested @decentralauren if she remembers discussing this and if that is the reason.

else I don't know why I lowered it

File: packages/app/src/components/DonateComponent.tsx [hunk 2]
```diff
@@ -201,521 +358,301 @@ function DonateComponent({ collective }: DonateComponentProps) {
...
+    (v: string, altDuration?: number) => {
+      const calculateEstDuration = (value: number, rate: number) => value / (rate === 0 ? 1 : rate);
+
+      let estDuration = altDuration ?? 0;
+
+      if (frequency === Frequency.Monthly && !altDuration) {
+        if (currency.includes('G$')) {
+          estDuration = parseFloat(donorCurrencyBalance) / parseFloat(v);
+        } else {
+          estDuration = calculateEstDuration(parseFloat(v), parseFloat(streamRate.toString()));
```
[the remark this answers is not in the document]
Comment (L03TJ3) [reply packages/app/src/components/DonateComponent.tsx:382]: 1. the duration is set based on a user's chosen streamrate of (amount / rate), and that is what we show in the UI
image

2. used when changing rate. other option is adding a flag `isRate` and moving the estDuration calculation from `onChangeRate` as alternative in the estimateDuration callback

File: packages/app/src/components/DonateComponent.tsx [hunk 3]
```diff
@@ -201,521 +358,301 @@ function DonateComponent({ collective }: DonateComponentProps) {
...
+                    </Text>
+                    <Text variant="bold" color="goodPurple.500" fontSize="lg">
+                      {estimatedDuration.endDate}
+                    </Text>
+                  </HStack>
+                </VStack>
+              ) : null}
+            </VStack>
+          </HStack>
+          {frequency !== 'One-Time' && currency === 'CELO' && isNonZeroDonation && swapValue ? (
```
[the remark this answers is not in the document]
Comment (L03TJ3) [reply packages/app/src/components/DonateComponent.tsx:599]: right, we want but forget it to add it where its supposed to.
added here: https://github.com/GoodDollar/GoodCollective/pull/232/commits/3395fa7a163b8af9d79e6bf1375804ca4db1a6a1

**Comment by sirpy** [comment]
> @L03TJ3 please attached images of different states

@L03TJ3

**Comment by L03TJ3** [comment]
@sirpy they have been added, do you see missing anything?

File: packages/app/src/models/constants.ts [hunk 4]
```diff
@@ -3,7 +3,7 @@ import GdContracts from '@gooddollar/goodprotocol/releases/deployment.json';
...
+export const acceptablePriceImpact = 2;
```
Comment (L03TJ3) [inline packages/app/src/models/constants.ts:6]:
```suggestion
export const acceptablePriceImpact = 5;
```

**Pull request merged**